app/models/metric/ci_mixin/capture.rb

```diff
@@ -80,10 +80,12 @@ def perf_capture_queue(interval_name, options = {})
       :state       => ['ready', 'dequeue'],
     }
 
+    messages = MiqQueue.where(queue_item).index_by { |m| [m.method_name, m.args] }
     items.each do |item_interval, *start_and_end_time|
       # Should both interval name and args (dates) be part of uniqueness query?
       queue_item_options = queue_item.merge(:method_name => "perf_capture_#{item_interval}")
       queue_item_options[:args] = start_and_end_time if start_and_end_time.present?
+      next if messages[queue_item_options.values_at(:method_name, :args)]
       MiqQueue.put_or_update(queue_item_options) do |msg, qi|
```

Comment:
When you put an item into the queue, should you also be updating your in-memory index?